ui-v2/app/styles/components/main-nav-horizontal/layout.scss

```diff
@@ -0,0 +1,135 @@
+%main-nav-horizontal > ul > li,
+%main-nav-horizontal-drop-nav a {
+  position: relative;
+}
+%main-nav-horizontal [type='checkbox'] ~ div {
+  display: none;
+}
+%main-nav-horizontal [type='checkbox']:checked ~ div {
+  display: block;
+}
+%main-nav-horizontal-action {
+  display: block;
+  padding: 3px 12px;
+  white-space: nowrap;
+}
+%main-nav-horizontal-drop-nav {
+  z-index: 400;
+  /* TODO: We should probably make menu-panel default to left hand side*/
+  left: 0;
+  right: auto;
+}
+%main-nav-horizontal-drop-nav header {
+  padding: 0.2rem 1rem;
+  padding-top: 0.7rem;
+}
+%main-nav-horizontal-drop-nav div {
+  padding: 10px;
+  padding-left: 36px;
```

Comment:
any reason for mixing `rem` and `px` and `em` in this file? would be nice to keep it to one unit if possible, but I realize odd px values causes weird decimals sometimes in rems/ems .

Reply:
'Generally' I try and use `em` for tops and bottoms, and then percentages for lefts/rights. That way you keep some sort of vertical rhythm (hence the top/bottom ems) and percentage/window width based based left/right responsiveness.

Consul UI isn't super strict with this, just due to the way we initially built it, and we never really got around to doing proper percentage based responsiveness so there is a lot of `px` and there is no spec for the vertical rhythm.

I'm guessing these rems are just copied over from what we had before here. I am intrigued as to why we went with `rem` here and not `em` though. I'll have a look and see, if its just what we had before though I'd keep it the same for the moment. All of the CSS changes here were trying to avoid changing things and just organizing/refactoring.

Reply:
Ah so just took a quick look, this is definitely new. I think it's related to text and being able to judge the width of the menu that we use that `@supports` for. I'm gonna take another look though as I'm wondering why I went with `rem` and not `em` 👍

Reply:
OMG - so, this is such an amazing deep dive! 😆 

Recently we made these things called `%internal-button`s in https://github.com/hashicorp/consul/pull/6495/files#diff-9ab79d006f089e09cd9e901c83788d3fR41 which are basically buttons within menu panels (as they don't use the border/background color the same as other buttons), and I traced the `rem` usage back to these (you can see the `0.75rem 1rem` in that PR link above).

The `rem`s you see in this file reuse the `1rem` left and right padding of these buttons as the left and right padding of the header needs to be the same as the `%internal-button` s below it:

![Screenshot 2019-12-11 at 17 28 40](https://user-images.githubusercontent.com/554604/70644858-ba273000-1c3b-11ea-84f2-6970149f5cc7.png)

See there ^ how the left edges of the buttons and the header (`NAMESPACES`) all line up exactly.

The top and bottom padding will have been 'make this look like the designs' which is probably why we use the `0.2rem` and `0.7rem` for top and bottom, I probably just tweaked this until it looked right/lined up with the designs.

So the big question is, why `rem`'s, I wouldn't have usually chosen those?

So the `%internal-buttons` came about when we were moving our menu styling to more closely follow structure _and_ while I was making sure they would work with @joshuaogle s slide-y confirmation dialogs in the menus, so followed the codepen for these as closely as possible - which was what made me think I wonder what the units where in that codepen? /me _come on browser history don't let me down_

![Screenshot 2019-12-11 at 17 32 40](https://user-images.githubusercontent.com/554604/70645132-49344800-1c3c-11ea-9c8e-ecfc0efaa3ba.png)

`0.75rem 1rem` ! Amazing!

(Sorry maybe it just me that loves this 😆 )

I'll probably find the `em`/`px` equivalent at some point in another PR, I'm just gonna drop a TODO comment in for now.